src/main/java/hudson/plugins/git/extensions/impl/CloneOption.java

```diff
@@ -20,13 +20,19 @@
  * @author Kohsuke Kawaguchi
  */
 public class CloneOption extends GitSCMExtension {
-    private boolean shallow;
-    private String reference;
-    private Integer timeout;
+    private final boolean shallow;
```

Comment:
CloneOption is used as a [value object](http://martinfowler.com/bliki/ValueObject.html). Its attributes are set once by the constructor (or the object is deserialized from xml file) and never changed again.
Making them final (and CloneOption immutable) makes reading and understanding the code easier. You simply don't have to think about what happens if one of these values changes (especially in a multi threading environment). It just doesn't happen. The compiler checks it for you.